diff --git a/fesm2022/http.mjs b/fesm2022/http.mjs
index af81338d92007ab22513e13290af306b495f4bfa..675fd01b9e537eee69cd7a4359ad270da095b8bd 100644
--- a/fesm2022/http.mjs
+++ b/fesm2022/http.mjs
@@ -79,6 +79,7 @@ class HttpResourceImpl extends _ResourceImpl {
     progress = this._progress.asReadonly();
     statusCode = this._statusCode.asReadonly();
     constructor(injector, request, defaultValue, parse, equal) {
+        const client = injector.get(HttpClient);
         super(request, ({ request, abortSignal }) => {
             let sub;
             // Track the abort listener so it can be removed if the Observable completes (as a memory
@@ -94,7 +95,7 @@ class HttpResourceImpl extends _ResourceImpl {
                 resolve?.(stream);
                 resolve = undefined;
             };
-            sub = this.client.request(request).subscribe({
+            sub = client.request(request).subscribe({
                 next: (event) => {
                     switch (event.type) {
                         case HttpEventType.Response:
@@ -129,7 +130,7 @@ class HttpResourceImpl extends _ResourceImpl {
             });
             return promise;
         }, defaultValue, equal, injector);
-        this.client = injector.get(HttpClient);
+        this.client = client;
     }
 }
 
